datafusion/physical-plan/src/joins/sort_merge_join.rs

```diff
@@ -852,6 +852,54 @@ fn get_corrected_filter_mask(
             corrected_mask.extend(vec![Some(true); null_matched]);
             Some(corrected_mask.finish())
         }
+        JoinType::Full => {
+            let mut mask: Vec<Option<bool>> = vec![Some(true); row_indices_length];
```

Comment:
Could use booleanbuilder?

Reply:
unlike to other join types its needed to update current array, the builder does append only afaik